### PR TITLE
Refactor more into protocol package to enable server-side code leverage

### PR DESCRIPTION
**Pull request by itzg**
* also cascaded to refactoring config to its own package

This will prepare us for a mini-endpoint server.

cc @rphillips

### CODE REVIEW

File: check/check.go [hunk 1]
```diff
@@ -109,6 +112,39 @@ func NewCheck(rawParams json.RawMessage) Check {
...
+	req.Version = "1"
+	req.Method = "check_metrics.post"
+	req.Params.EntityId = crs.Check.GetEntityId()
+	req.Params.CheckId = crs.Check.GetId()
+	req.Params.CheckType = crs.Check.GetCheckType()
+	req.Params.Metrics = []protocol.MetricWrap{ConvertToMetricResults(crs)}
+	req.Params.MinCheckPeriod = crs.Check.GetPeriod() * 1000
+	req.Params.State = crs.State
+	req.Params.Status = crs.Status
+	req.Params.Timestamp = utils.NowTimestampMillis()
```
Comment (jjbuchan) [inline check/check.go]: Note that even though the agent sends the timestamp, I'm pretty sure we'll still be ignoring it in the AEP.

**Comment by jjbuchan** [comment]
Tentative +1.  I don't have much of an opinion on these right now, since I haven't worked with the poller code, but seems fine to me.

File: check/check.go [hunk 2]
```diff
@@ -109,6 +112,39 @@ func NewCheck(rawParams json.RawMessage) Check {
...
 
+func NewMetricsPostRequest(crs *CheckResultSet) *protocol.MetricsPostRequest {
+	req := &protocol.MetricsPostRequest{}
+	req.Version = "1"
+	req.Method = "check_metrics.post"
+	req.Params.EntityId = crs.Check.GetEntityId()
+	req.Params.CheckId = crs.Check.GetId()
+	req.Params.CheckType = crs.Check.GetCheckType()
+	req.Params.Metrics = []protocol.MetricWrap{ConvertToMetricResults(crs)}
+	req.Params.MinCheckPeriod = crs.Check.GetPeriod() * 1000
```
Comment (rphillips) [inline check/check.go]: thinking this should live in a /messages package or within messages.go, so all messages are in one spot

Comment (itzg) [inline check/check.go]: I'll look again but I believe the CheckResultSet usage would have caused a package-cycle.

Comment (itzg) [inline check/check.go]: I'm going to try shuffling just the struct's into the protocol package since that would be necessary for poller-agent cross use.

**Comment by rphillips** [comment]
+1

**Comment by itzg** [comment]
Addresses #13

**Comment by itzg** [comment]
@rphillips this is ready for a re-review with the JSON bits of hostinfo moved into a new `protocol/hostinfo` package

**Comment by rphillips** [comment]
+1

**Pull request merged**